csrc/ir/builder.cpp

```diff
@@ -330,6 +332,9 @@ Val* SimplifyingIrBuilder::addExpr(Val* lhs, Val* rhs) {
 }
 
 Val* SimplifyingIrBuilder::subExpr(Val* lhs, Val* rhs) {
+  if (lhs->sameAs(rhs)) {
```

Comment:
What if `lhs` is NAN?

Reply:
Ah good point. I can guard this with `lhs->isIntegralScalar()` then it's valid right?

Reply:
I think so